Mathlib/MeasureTheory/Integral/IntervalIntegral.lean

```diff
@@ -373,6 +373,64 @@ theorem Antitone.intervalIntegrable {u : ℝ → E} {a b : ℝ} (hu : Antitone u
 
 end
 
+/-!
+## Interval integrability of functions with even or odd parity
+-/
+section
+
+variable {f : ℝ → E}
+
+/-- An even function is interval integrable (with respect to the volume measure)
+on every interval if it is interval integrable (with respect to the volume
+measure) on every interval of the form `0..x`, for positive `x`. -/
+
+theorem intervalIntegrable_of_even
+  (h₁f : ∀ x, f x = f (-x))
+  (h₂f : ∀ x, 0 < x → IntervalIntegrable f volume 0 x) :
+  ∀ x y, IntervalIntegrable f volume x y := by
+  -- Lemma: Prove statement first in case where x = 0
+  have : ∀ t, IntervalIntegrable f volume 0 t := by
+    intro t
+    rcases lt_trichotomy t 0 with h|h|h
+    · rw [IntervalIntegrable.iff_comp_neg]
+      conv => arg 1; intro t; rw [← h₁f]
+      simp [h₂f (-t) (by norm_num [h])]
+    · rw [h]
+    · exact h₂f t h
+  -- Split integral and apply lemma
+  intro x y
+  exact IntervalIntegrable.trans (b := 0) (IntervalIntegrable.symm (this x)) (this y)
+
+
+/-- An odd function is interval integrable (with respect to the volume measure)
+on every interval iff it is interval integrable (with respect to the volume
+measure) on every interval of the form `0..x`, for positive `x`. -/
+
+theorem intervalIntegrable_of_odd
+  {f : ℝ → ℝ}
+  (h₁f : ∀ x, -f x = f (-x))
+  (h₂f : ∀ x, 0 < x → IntervalIntegrable f volume 0 x) :
+  ∀ x y, IntervalIntegrable f volume x y := by
+  -- Lemma: Prove statement first in case where x = 0
+  have : ∀ t, IntervalIntegrable f volume 0 t := by
+    intro t
+    rcases lt_trichotomy t 0 with h|h|h
+    · rw [IntervalIntegrable.iff_comp_neg]
+      conv => arg 1; intro t; rw [← h₁f]
+      apply IntervalIntegrable.neg
+      simp [h₂f (-t) (by norm_num [h])]
+    · rw [h]
+    · exact h₂f t h
+  -- Split integral and apply lemma
+  intro x y
+  exact IntervalIntegrable.trans (b := 0) (IntervalIntegrable.symm (this x)) (this y)
```

Comment:
Do you know about dot notation already? This allows writing proofs like this more concisely, for instance (untested)
```suggestion
exact (this x).symm.trans (b := 0) (this y)

Reply:
Then you could even combine this with the previous line, into `exact fun x y ↦ (this x).symm.trans (b := 0) (this y)`